solidity/contracts/middleware/liquidity-layer/adapters/CctpAdapter.sol

```diff
@@ -0,0 +1,185 @@
+    ITokenMessenger public tokenMessenger;
+
+    /// @notice The token address.
+    address public token;
+
+    /// @notice The token symbol of the token.
+    string public tokenSymbol;
+
+    /// @notice The amount of gas required to process a CCTP transfer.
+    uint256 public gasAmount;
+
+    /// @notice Hyperlane domain => Circle domain.
+    /// ATM, known Circle domains are Ethereum = 0 and Avalanche = 1.
+    /// Note this could result in ambiguity between the Circle domain being
+    /// Ethereum or unknown.
+    mapping(uint32 => uint32) public hyperlaneDomainToCircleDomain;
+
+    /**
+     * @notice Emits the nonce of the Circle message when a token is bridged.
+     * @param nonce The nonce of the Circle message.
+     */
+    event BridgedToken(uint64 nonce);
+
+    /**
+     * @notice Emitted when the Hyperlane domain to Circle domain mapping is updated.
+     * @param hyperlaneDomain The Hyperlane domain.
+     * @param circleDomain The Circle domain.
+     */
+    event DomainAdded(uint32 indexed hyperlaneDomain, uint32 circleDomain);
+
+    /**
+     * @param _owner The new owner.
+     * @param _tokenMessenger The TokenMessenger contract.
+     * @param _token The token address.
+     * @param _tokenSymbol The token symbol.
+     * @param _gasAmount The amount of gas required to process a CCTP transfer.
+     * @param _mailbox The address of the mailbox contract.
+     * @param _interchainGasPaymaster The address of the interchain gas paymaster contract.
+     * @param _interchainSecurityModule The address of the interchain security module contract.
+     */
+    function initialize(
+        address _owner,
+        address _tokenMessenger,
+        address _token,
+        string calldata _tokenSymbol,
+        uint256 _gasAmount,
+        address _mailbox,
+        address _interchainGasPaymaster,
+        address _interchainSecurityModule
+    ) external initializer {
+        __HyperlaneConnectionClient_initialize(
+            _mailbox,
+            _interchainGasPaymaster,
+            _interchainSecurityModule,
+            _owner
+        );
+
+        tokenMessenger = ITokenMessenger(_tokenMessenger);
+        token = _token;
+        tokenSymbol = _tokenSymbol;
+        gasAmount = _gasAmount;
+    }
+
+    /**
+     * @notice Transfers `_amount` token to `_recipientAddress` on `_destinationDomain` chain.
+     * @param _destinationDomain The identifier of the destination chain.
+     * @param _recipientAddress The address of the recipient on the destination chain.
+     * @param _amount The amount of tokens to transfer.
+     * @return messageId The identifier of the dispatched message.
+     */
+    function transferRemote(
+        uint32 _destinationDomain,
+        bytes32 _recipientAddress,
+        uint256 _amount
+    ) external payable override returns (bytes32 messageId) {
+        _mustHaveRemoteRouter(_destinationDomain);
+        uint32 _circleDomain = hyperlaneDomainToCircleDomain[
```

Comment:
Isn't it possible to return 0 for non-configured domains?

 If so, can we add a non-zero check and maybe add 1 to circle mapping to remove this ambiguity?

Reply:
Seems safer but I think it would pose a maintenance risk for devs when they add new domains (when cctp adds new domains) and they fail to add 1 to the circle mapping?